frontend/src/pages/AnnouncementPage.js

```diff
@@ -0,0 +1,420 @@
+import Collapse from "@mui/material/Collapse";
+import { Link, useLocation } from "react-router-dom";
+import RoundedEdgesButton from "../components/RoundedEdgesButton";
+
+function InputFields({ defaultTitle, defaultContent, titleComponenId, contentComponenId }) {
+	// TODO: formated text support
+	// ref: https://mui.com/components/text-fields/#integration-with-3rd-party-input-libraries
+	return <Grid>
+		<Grid item xs={12}>
+			<TextField
+				required
+				fullWidth
+				name="title"
+				label="Title"
+				id={titleComponenId}
+				defaultValue={defaultTitle}
+				inputProps={{ style: { color: "white" } }}
+				multiline
+				minRows={1}
+				maxRows={10}
+				sx={{ multilineColor: { color: "white" } }}
+			/>
+		</Grid>
+		<Grid item xs={12}>
+			<TextField
+				required
+				fullWidth
+				name="content"
+				label="Content"
+				id={contentComponenId}
+				defaultValue={defaultContent}
+				inputProps={{ style: { color: "white" } }}
+				multiline
+				minRows={1}
+				maxRows={10}
+			/>
+		</Grid>
+	</Grid>;
+}
+
+function AnnouncementEntry({ 
+	announcement, editing, setEditing, userIsAdmin, setAnnouncementsAreUpToDate, isSidebar 
+}) {
+	const created_at = new Date(announcement.created_at * 1000);
+	const formated_created_at =
+		created_at.getDate() + "/" +
+		created_at.getMonth() + "/" +
+		created_at.getFullYear() + " " +
+		("0" + created_at.getHours()).slice(-2) + ":" + // Leading zeroes
+		("0" + created_at.getMinutes()).slice(-2);
+
+	const handleEditButton = id => () => {
+		setEditing(id);
+	};
+
+	const changeRoute = useChangeRoute();
+
+	const handleDeleteButton = id => () => {
+		appwriteApi.deleteAnnouncement(id)
+			.then(() => {
+				setAnnouncementsAreUpToDate(false);
+			})
+			.catch((e) => {
+				console.log(e);
+			});
+	};
+
+	const handleCancelButton = () => {
+		setEditing("");
+		if (window.location.hash.split("#")[1]) {
+			changeRoute("/announcements");
+		}
+	};
+
+	const handleSubmitButton = (announcementId, titleComponenId, contentComponenId) => () => {
+		const title = document.getElementById(titleComponenId);
+		const content = document.getElementById(contentComponenId);
+		if (title.value.length == 0 || content.value.length == 0) {
+			console.log("missing input or content!");
+			return;
+		}
+		appwriteApi.updateAnnouncement({
+			"title": title.value,
+			"content": content.value
+		}, announcementId)
+			.then(() => {
+				setAnnouncementsAreUpToDate(false);
+				if (window.location.hash.split("#")[1]) {
+					changeRoute("/announcements");
+				}
+			})
+			.catch((e) => {
+				console.log(e);
+			});
+	};
+
+	return <div style={{ width: "100%" }}>
+		{isSidebar
+			?
+			<Box sx={{ display: "flex", p: 1, bgcolor: "blue" }}>
+				<Container>
+					<Typography variant="h5" >{announcement.title.substring(0, 50) + "..."}</Typography >
+					<Typography><i>{formated_created_at}</i></Typography >
+					<Typography>{announcement.title.substring(0, 100) + "..."}</Typography >
```

Comment:
Here you probably meant this
```suggestion
					<Typography>{announcement.content.substring(0, 100) + "..."}</Typography >
```